### DISCUSSION TITLE
MQTT - Can't communicate in mqtt (MqttPersistenceException)

### DISCUSSION THREAD

**Discussion by fabiodurao** [SCADA-LTS/Scada-LTS#2617] [General]
I was trying to upload a slts container to docker (thanks again for the help) to test mqtt and try to figure out what might be going on. I'm not getting communication in mqtt on slts 2.7.5.3 installed on aws on ubuntu 22.04.
I tested the same communication on slts 2.7.5.3 on windows on my pc (installation with linux installer), on linux on my pc (installation with docker), on linux on a remote server (installation with linux installer) and I had success in communication on all these scenarios (I also tested with mqtt explorer and everything was fine), but in this installation that is on AWS with ubuntu 2.7.5.3, with manual installation, I am receiving the alarm below and there is no value in the watch list.

From what I researched and understood, it seems that the server refuses communication for having two connections, I didn't understand very well, I already deleted all the mqtt datasources and recreated them, I even tried to open port 1883 at the entrance of the firewall, which doesn't make much sense, but I don't know what it is, do you have any idea how I can solve this?

14:49:35:	'test_ds': org.scada_lts.ds.messaging.exception.MessagingServiceException: Error Init Receiver: datasource: test_ds (id: 7, xid: DS_618510, type: MQTT), exception: MessagingChannelException (msg: Error Create Channel: datapoint: test (id: 203, xid: DP_585595, dataSourceId: 7, dataSourceName: test_ds), exception: MqttPersistenceException (msg: MqttException))

**Comment by Limraj** [comment]
Do you have a broker running?

Here you have the configuration in docker-compose, along with the mqtt broker:
https://github.com/SCADA-LTS/Scada-LTS/pull/2571/files


You need to remove the following lines:
          - ./tomcat/webapps:/usr/local/tomcat/webapps:rw
          - JPDA_ADDRESS=0.0.0.0:8000
          - JPDA_TRANSPORT=dt_socket
          - jpda

**Comment by fabiodurao** [comment]
In this installation that I just did with docker and your help, mqtt worked perfectly. There is a broker on the server where I am pointing the datasource, I was able to bring data from there using mqtt explorer and Scada-LTS in different types of installation, windows and linux, local and remote.

What is not working, is the mqtt only in the slts 2.7.5.3 installation that I did manually (https://github.com/SCADA-LTS/Scada-LTS/wiki/Scada-LTS-Linux-installation-tutorial) in ubuntu 22.04 on aws. On this server I already use other datasources in other protocols normally.

If I copy from the mqtt datasource, the host, username, password from and copy the "topic filter" from the datapoint, to another slts of those that I mentioned earlier, it works. I'm not copying the "client ID" so it doesn't get duplicated.

I believe that these tips you gave me would be in case the failure is occurring in the installation with docker, right? I just used docker to validate if the problem was punctual or was in other forms of installation or operating system too.

In short, this same test works on
- windows -installer
- linux - installer
- linux - docker

Does not work
- linux - manual

but I don't know if the manual installation is the variable of interest to solve this problem.

**Comment by Limraj** [comment]
@fabiodurao, this manual has been updated and there is a description that it should be kept as an archive, but there are links to customer and developer installation instructions create by @Patrykb0802. Okay, I'm closing it.
